addons/point_of_sale/static/src/js/models.js

```diff
@@ -211,7 +211,7 @@ exports.PosModel = Backbone.Model.extend({
             });
             _.each(self.taxes_by_id, function(tax) {
                 tax.child_taxes = {};
-                _.each(tax.child_ids, function(child_tax_id) {
+                _.each(tax.children_tax_ids, function(child_tax_id) {
                     tax.child_taxes[child_tax_id] = self.taxes_by_id[child_tax_id];
                 });
```

Comment:
We should try and keep the POS accounting stuff as consistent as possible with that of `account.py`. It's already confusing enough as it is. So a tax in the POS should have `children_taxes_ids` that is an array of other account.tax references. So essentially all you need to do is replace the array of ids with references to the correct taxes. Something like:

```javascript
_.each(self.taxes_by_id, function(tax) {
    // children_tax_ids is a m2m, which get loaded as an
    // array of ids, this just replaces the ids with the
    // corresponding account.tax references
    tax.children_tax_ids = _.map(tax.children_tax_ids, function (child_tax_id) {
        return self.taxes_by_id[child_tax_id];
    });
});
```

would be nice.